Shoko.TrayService/App.xaml.cs

```diff
@@ -1,174 +1,167 @@
-            string instance = null;
-            for (int x = 0; x < e.Args.Length; x++)
-            {
-                if (!e.Args[x].Equals("instance", StringComparison.InvariantCultureIgnoreCase)) continue;
-                if (x + 1 >= e.Args.Length) continue;
-                instance = e.Args[x + 1];
-                break;
-            }
-
-            var arguments = new ProgramArguments {Instance = instance};
-            if (!string.IsNullOrEmpty(arguments.Instance)) ServerSettings.DefaultInstance = arguments.Instance;
+    private void OnStartup(object a, StartupEventArgs e)
+    {
+        Console.CancelKeyPress += OnConsoleOnCancelKeyPress;
+        InitialiseTaskbarIcon();
+        StartServer.StartupServer(AddEventHandlers, ServerStart);
+    }
 
-            ShokoServer.Instance.InitLogger();
-            
-            ServerSettings.LoadSettings();
-            ServerState.Instance.LoadSettings();
+    private static bool ServerStart() 
+        => ShokoServer.Instance.StartUpServer();
 
-            if (!ShokoServer.Instance.StartUpServer()) return;
+    private void AddEventHandlers()
+    {
+        ShokoServer.Instance.ServerShutdown                       += OnInstanceOnServerShutdown;
+        Utils.YesNoRequired                                       += OnUtilsOnYesNoRequired;
+        ServerState.Instance.PropertyChanged                      += OnInstanceOnPropertyChanged;
+        ShokoService.CmdProcessorGeneral.OnQueueStateChangedEvent += OnCmdProcessorGeneralOnOnQueueStateChangedEvent;
+    }
+    
+    private void OnCmdProcessorGeneralOnOnQueueStateChangedEvent(QueueStateEventArgs ev) 
+        => Console.WriteLine($"General Queue state change: {ev.QueueState.formatMessage()}");
+    
+    private static void OnUtilsOnYesNoRequired(object? _, Utils.CancelReasonEventArgs e) 
+        => e.Cancel = true;
+    
+    private void OnInstanceOnServerShutdown(object? o, EventArgs eventArgs) 
+        => Shutdown();
 
-            if (!ServerSettings.Instance.FirstRun)
-                ShokoServer.RunWorkSetupDB();
-            else Logger.Warn("The Server is NOT STARTED. It needs to be configured via webui or the settings.json");
+    private static void OnInstanceOnPropertyChanged(object? _, PropertyChangedEventArgs e)
+    {
+        if (e.PropertyName == "StartupFailedMessage" && ServerState.Instance.StartupFailed)
+            Console.WriteLine("Startup failed! Error message: " + ServerState.Instance.StartupFailedMessage);
+    }
 
-            ShokoServer.Instance.ServerShutdown += (sender, eventArgs) => Shutdown();
-            Utils.YesNoRequired += (sender, e) =>
-            {
-                e.Cancel = true;
-            };
+    private void InitialiseTaskbarIcon()
+    {
+        _icon = new TaskbarIcon{
+                                   ToolTipText = "Shoko Server", ContextMenu = CreateContextMenu(), MenuActivation = PopupActivationMode.All, Visibility = Visibility.Visible,
+                               };
```

Comment:
If you're going to make this a multi-line object constructor, do each property as it's own line

Reply:
please provide a complete .editorconfig for fomating issues, this is ReSharper default / detection of Shoko's Style.

Reply:
That will be in the works but in the mean time, you can change this.